### ISSUE TITLE
perl-X11-GUITest and Fedora 33 - unable to install the Asbru-CM

### ISSUE THREAD

**Issue by danielbgto** [asbru-cm/asbru-cm#707]
HI! I'm thinking this is not a bug to Asbru-cm but a Fedora team move. So... long story - shot: I have tried install Asbru-cm  on Fedora 33 Beta machine and I get that:
- nothing provides perl-X11-GUITest

Here I found that with Fedora 33 some packages are marked obsolete, including the needed one  "perl-X11-GUITest" https://bugzilla.redhat.com/show_bug.cgi?id=1884617

Kindly asking Is there way to solve that? I'm using Fedora and Asbru-cm every day 

Some output  trying the all main asbru versions. Getting the same error:

[pavitra@f33-test ~]$ sudo dnf install asbru-cm-6.2.1-1.fc32.noarch.rpm
Last metadata expiration check: 0:28:09 ago on Fri 16 Oct 2020 08:58:04 PM CEST.
Error: 
 Problem: conflicting requests
  - nothing provides perl-X11-GUITest needed by asbru-cm-6.2.1-1.fc32.noarch
(try to add '--skip-broken' to skip uninstallable packages)

[pavitra@f33-test ~]$ sudo dnf install asbru-cm-6.1.4~1589485565-1.fc31.noarch.rpm
Last metadata expiration check: 0:28:27 ago on Fri 16 Oct 2020 08:58:04 PM CEST.
Error: 
 Problem: conflicting requests
  - nothing provides perl-X11-GUITest needed by asbru-cm-6.1.4~1589485565-1.fc31.noarch
(try to add '--skip-broken' to skip uninstallable packages)

[pavitra@f33-test ~]$ sudo dnf install asbru-cm-5.2.1~1581754379-1.fc30.noarch.rpm
Last metadata expiration check: 0:28:43 ago on Fri 16 Oct 2020 08:58:04 PM CEST.
Error: 
 Problem: conflicting requests
  - nothing provides perl-X11-GUITest needed by asbru-cm-5.2.1~1581754379-1.fc30.noarch
(try to add '--skip-broken' to skip uninstallable packages)

**Comment by gfrenoy** [comment]
I believe this package should be optional ... You may try the suggested option `--skip-broken`.

It is used to better position the RDP embed windows.  But if not present, it should not prevent Ásbrú to work.

**Comment by popxunga** [comment]
Hi @danielbgto ...
While I'm not sure if this will work or not, you may try grab the fc32 perl-X11-GUITest from [here](https://download-ib01.fedoraproject.org/pub/fedora/linux/releases/32/Everything/x86_64/os/Packages/p/perl-X11-GUITest-0.28-19.fc32.x86_64.rpm), and then install it with: sudo dnf install perl-X11-GUITest-0.28-19.fc32.x86_64.rpm.
Finally attempt the Asbru installation.
Another option would be built the package from source.
Good luck.

**Comment by danielbgto** [comment]
Thank you for the help! I got it working via _**very dirty hack**_. It's not recommended to production environment!  

1. Installed the **perl-X11-GUITest** from F32 with sudo rpm -i perl-X11-GUITest-0.28-19.fc32.x86_64.rpm --force --nodeps  // this will trick the system that perl-X11-GUITest is already installed and will give us the road to:
2.  sudo dnf install asbru-cm-5.2.1~1581754379-1.fc30.noarch.rpm // I prefer  5.2.1 because of the priceless command history // The installation will be ok but asbru will not actually work. It will start but not able to make a session.
3.  sudo dnf install libXt-devel.x86_64 libXtst-devel.x86_64 gtk-murrine-engine.x86_64 // compile dependencies and murrine engine 
4. Compile perl-X11-GUITest from the source - tar file here https://sourceforge.net/projects/x11guitest/ and installation instructions here - https://metacpan.org/pod/X11::GUITest
5. Start the Asbru

**Comment by danielbgto** [comment]
I have compiled perl-X11-GUITest source rpm package against Fedora 33. This is the clearest way IMHO.
The package could be downloaded (the link below) and installed before Asbru package installation. Just tested it on fresh installed F33 VM.  
https://drive.google.com/file/d/16hJR2mGa8OuHhvauk9S9LnMe5RIWCts0/view?usp=sharing

Fedora and Asbru fans are again happy together :)))

**Comment by gfrenoy** [comment]
Still believe this package is not mandatory so I made it a "suggested" package on the loki branch.  At least it should not prevent the Ásbrú installation if this package is not provided by the main distribution.

**Comment by popxunga** [comment]
I'm a bit confused I should say.
Can't we check if the Asbru code uses or not functions provided by that package ?

**Comment by gfrenoy** [comment]
Yes, search for "GUITest" ; you'll see it's optional if the package is installed (or not) ... It was more a hack to workaround some other issues, it can help in some cases but it's not mandatory for sure.  I did uninstall the package and did not spot any issue so far.

**Comment by popxunga** [comment]
And are you testing with your WM in Wayland or X11 mode ? 
Do we have any history on what those issues were ?

Also keep in mind that the "Suggested" keyword migth not be supported on all distros make engines.

**Comment by gfrenoy** [comment]
See #153

**Comment by danielbgto** [comment]
I did the same test with removing GUITest. After removing it Asbru is working on F33. The thing what mislead me is that I had installed F32 package and then testing Asbru working but opening the terminal was giving problem(s) with "GUITest.pm" but this is because the libraries are against other version of Perl.... No screenshot from that moment of my first tests. I needed to delete that file and should work properly on that time but I had no mind for that at those moment. Now I understood the logic fully.
Anyway ... sorry misleading you all. The package looks like "optional" for some features and definitely is impossible to install from Fedora 33.
